### PR TITLE
Diagnose resources that conflict with the resource bundle Info.plist

### PR DESCRIPTION
**Pull request by hartbit**

### CODE REVIEW

File: Sources/PackageLoading/Diagnostics.swift [hunk 1]
```diff
@@ -77,6 +77,16 @@ extension Diagnostic.Message {
...
         .note("found '\(path)'")
     }
+
+    static func infoPlistResourceConflict(
+        path: RelativePath,
+        targetName: String
+    ) -> Self {
+        .error("""
+            resource '\(path)' in target '\(targetName)' is forbidden as it conflicts with the generated resource \
+            bundle Info property list
```
Comment (aciidb0mb3r) [inline Sources/PackageLoading/Diagnostics.swift]: We shouldn't talk about the generated resource bundle. How about something along the lines of "Info.plist is not supported as a resource file"

Comment (hartbit) [reply Sources/PackageLoading/Diagnostics.swift]: I'm worried that may be confusing: Info.plist is not forbidden - it's Info.plist at root which is. Do you have another suggestion?

Comment (aciidb0mb3r) [reply Sources/PackageLoading/Diagnostics.swift]: Hmm, "Info.plist is not supported as a top-level resource file in the resources bundle"

Comment (hartbit) [reply Sources/PackageLoading/Diagnostics.swift]: Updated!

**Comment by hartbit** [comment]
@swift-ci smoke test

**Pull request merged**